packages/calcite-components/src/components/link/link.scss

```diff
@@ -63,7 +63,7 @@ calcite-icon {
 }
 
 :host {
-  span,
+  button,
```

Comment:
Suggestion: styles and tests could be simplified if we use a `link` class and use it for targeting both `button` and `a` elements. 

We can tackle this separately to keep the PR focused.